lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1797,7 +1660,7 @@ private function doPersist($entity, array &$visited): void
             case self::STATE_REMOVED:
```

Comment:
These switch/cases are needed to be replaced with state pattern (EntityStateManager class)